ERCS/erc-7562.md

```diff
@@ -0,0 +1,347 @@
+* **[EREP-030]** A Staked Account is accountable for failures in other entities (`paymaster`, `aggregator`) even if they are staked.
+* **[EREP-040]** An `aggregator` must be staked, regardless of storage usage.
+* **[EREP-050]** An unstaked `paymaster` may not return a `context`.
+    * TODO: can be removed once we remove the "2nd postOp".
+
+### Unstaked Entities Reputation Rules
+
+* Definitions:
+    * **`opsSeen`, `opsIncluded`, and reputation calculation** are defined above.
+    * `UnstakedReputation` of an entity is a maximum number of entries using this entity allowed in the mempool.
+    * `opsAllowed` is a reputation based calculation for an unstaked entity, representing how many `UserOperations` it is allowed to have in the mempool.
+* **[UREP-010]** `UserOperation` with unstaked sender are only allowed up to `SAME_SENDER_MEMPOOL_COUNT` times in the mempool.
+* **[UREP-020]** For other entities: \
+    `opsAllowed = SAME_UNSTAKED_ENTITY_MEMPOOL_COUNT + (inclusionRate * INCLUSION_RATE_FACTOR) + (min(opsIncluded, 10000)`.
+    * This is a default of `SAME_UNSTAKED_ENTITY_MEMPOOL_COUNT` for new entity
+* **[UREP-030]** If an unstaked entity causes an invalidation of a bundle, its `opsSeen` is set to `1000`, effectively blocking it from inclusion for 24 hours.
+
+### Alt-mempools rules:
+
+Alternate mempool is an agreed-upon rule that the bundlers may opt in to.
+
+* **[ALT-010]** The bundler listens to the alt-mempool "topic" over the P2P protocol
+* **[ALT-020]** The alt mempool rules MUST be checked only when a canonical rule is violated
+    * That is, if validation follows the canonical rules above, it is not considered part of an alt-mempool.
+* **[ALT-030]** Bundlers SHOULD forward `UserOperations` to other bundlers only once, regardless of how many (shared) alt-mempools they have. \
+The receiving bundler validates the `UserOperations`, and based on the above rules (and subscribed alt-mempools) decides for which alt-mempools to "tag" it.
+
+
+## Rationale
+
+All transactions initiated by EOAs have an implicit validation phase where balance, nonce, and signature are
+checked to be valid for the current state of the Ethereum blockchain.
+Once the transaction is checked to be valid by a node, only then another transaction by the same EOA can modify the Ethereum
+state in a way that makes the first transaction invalid.
+
+With Account Abstraction, however, the validation can also include an arbitrary EVM code and rely on storage as well,
+which means that unrelated `UserOperations` or transactions may invalidate each other.
+
+If not addressed, this would make the job of maintaining a mempool of valid `UserOperations` and producing valid
+bundles computationally infeasible and susceptible to DoS attacks.
+
+This document describes a set of validation rules that, if applied by a bundler before accepting a `UserOperation`
+into the mempool can prevent such attacks.
```

Comment:
This still sorta reads like it belongs in Motivation. The Rationale section should explain why technical choices were made within the EIP. A good example here would be why [UREP-020] uses the particular formula given above.